applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -416,6 +416,134 @@ bool CheckWakeCondition(const Element& rElement, const double& rTolerance, const
     return wake_condition_is_fulfilled;
 }
 
+template <int Dim, int NumNodes>
+double ComputeLocalMachSquaredDerivative(
+    const array_1d<double, Dim>& rVelocity,
+    const double localMachNumber,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // Following Fully Simulataneous Coupling of the Full Potential Equation
+    //           and the Integral Boundary Layer Equations in Three Dimensions
+    //           by Brian Nishida (1996), Section A.2
+
+    // read free stream values
+    const double heat_capacity_ratio = rCurrentProcessInfo[HEAT_CAPACITY_RATIO];
+    const double free_stream_mach = rCurrentProcessInfo[FREE_STREAM_MACH];
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+
+    // make squares of values
+    const double sq_local_mach_number = std::pow(localMachNumber, 2);
+    const double sq_free_stream_mach = std::pow(free_stream_mach, 2);
+    const double sq_free_stream_velocity = inner_prod(free_stream_velocity, free_stream_velocity);
+
+    // computes square of velocity including clamping according to MACH_SQUARED_LIMIT
+    const double sq_local_velocity = ComputeClampedVelocitySquared<Dim, NumNodes>(rVelocity, rCurrentProcessInfo);
+
+    // square bracket term
+    double square_bracket_term = 1.0 + 0.5*(heat_capacity_ratio - 1.0)*
+            sq_free_stream_mach*(1.0 - (sq_local_velocity/sq_free_stream_velocity));
+
+    // calculate mach derivative
+    return sq_local_mach_number * ((1.0/sq_local_velocity) +
```

Comment:
This one was easy to review cause you implemented it exactly as it comes in section A.2.5., thanks for that! :+1: 
As we talked, please think whether it is possible to make this line more readable (I know that sometimes it is difficult with such long formulas) (maybe splitting it into different lines?). Also check the formatting, I think there are some spaces missing, [clang](https://github.com/KratosMultiphysics/Kratos/wiki/How-to-configure-clang%E2%80%90format#clang-format-file) should automatically take care of that for you.